FfaFeaturePkg/Library/TpmServiceStateTranslationLib/TpmServiceStateTranslationLib.c

```diff
@@ -124,22 +126,30 @@ FifoReadBurstCount (
   UINT16                  *BurstCount
   )
 {
-  UINT32  Timeout;
-  UINT8   DataByte0;
-  UINT8   DataByte1;
+  EFI_STATUS  Status;
+  UINT32      DelayAmount;
+  UINT8       DataByte0;
+  UINT8       DataByte1;
 
-  Timeout = 0;
+  DelayAmount = 0;
   do {
+    MicroSecondDelay (DELAY_AMOUNT);
+
     DataByte0   = MmioRead8 ((UINTN)&ExternalFifo->BurstCount);
     DataByte1   = MmioRead8 ((UINTN)&ExternalFifo->BurstCount + 1);
     *BurstCount = (UINT16)((DataByte1 << 8) + DataByte0);
     if (*BurstCount != 0) {
       return EFI_SUCCESS;
     }
 
-    MicroSecondDelay (DELAY_AMOUNT);
-    Timeout += DELAY_AMOUNT;
-  } while (Timeout < PTP_TIMEOUT_D);
+    Status = ArmFfaLibYield (YIELD_AMOUNT);

```

Comment:
perhaps a nitpick, but it's best if we don't yield just to come back and timeout. Perhaps the timeout should be a break after the final failed call.